couchdb/server/__init__.py

```diff
@@ -61,6 +61,107 @@ def __init__(self, version=None, input=sys.stdin, output=sys.stdout,
         for key, value in options.items():
             self.handle_config(key, value)
 
+    def config_log_level(self, value):
+        """Sets overall logging level.
+
+        :param value: Valid logging level name.
+        :type value: str
+        """
+        log.setLevel(getattr(logging, value.upper(), 'INFO'))
+
+    def config_log_file(self, value):
+        """Sets logging file handler. Not used by default.
+
+        :param value: Log file path.
+        :type value: str
+        """
+        handler = logging.FileHandler(value)
+        handler.setFormatter(logging.Formatter(
+            '[%(asctime)s] [%(name)s] [%(levelname)s] %(message)s'
+        ))
+        log.addHandler(handler)
+
+    @property
+    def config(self):
+        """Proxy to query server configuration dictionary. Contains global
+        config options."""
+        return self._config
+
+    def handle_config(self, key, value):
+        """Handles config options.
+
+        Invoke the handler according to function name ``config_{key}``.
+
+        :param key: Config option name.
+        :type key: str
+
+        :param value:
+        """
+        handler_name = 'config_{0}'.format(key)
+        if hasattr(self, handler_name):
+            getattr(self, handler_name)(value)
+        else:
+            self.config[key] = value
+
+    def serve_forever(self):
+        """Query server main loop. Runs forever or till input stream is opened.
+
+        :returns:
+            - 0 (`int`): If :exc:`KeyboardInterrupt` exception occurred or
+              server has terminated gracefully.
+            - 1 (`int`): If server has terminated by
+            :py:exc:`~couchdb.server.exceptions.FatalError` or by another one.
+        """
+        try:
+            for message in self.receive():
```

Comment:
If you like, you can catch json decode exception and throw it as RuntimeError, catch it here before `except Exception` and reraise as is to throw whole stacktrace back to stderr, but not sure if all this worth any actions.